File: api/src/main/java/com/repeti/api/model/Alternativa.java
package com.repeti.api.model;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString

@Entity
@Table(name = "alternativa")
public class Alternativa {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String alternativa;

    private Boolean resposta;

    @ManyToOne
    // @JoinColumn(name = "questao_id", nullable = false)
    private Questao questao;
}

File: src/api/src/core/models/impl/questao/alternativa.ts
import { IAlternativa } from '../../interface/alternativa';
import { IQuestao } from '../../interface/questao';

export class AlternativaImpl implements IAlternativa {
  questao?: IQuestao;
  questaoId: number;
  id: number;
  descricao: string;
  resposta: boolean;
}
